src/util/std_expr.h

```diff
@@ -1414,6 +1415,29 @@ class equal_exprt:public binary_relation_exprt
     binary_relation_exprt(_lhs, ID_equal, _rhs)
   {
   }
+
+  static void check(
+    const exprt &expr,
+    const validation_modet vm = validation_modet::INVARIANT)
+  {
+    DATA_CHECK(expr.operands().size() == 2, "equality must have two operands");
+  }
+
```

Comment:
The validation_modet vm doesn't appear to be used? Can it just simply be removed?

Reply:
What is the reason for this _implicit_ use? Why can't we just pass it as another parameter?

Reply:
I just wanted to avoid having to pass the argument every time `DATA_CHECK` is used. But I can make it into a parameter if people prefer it.

Reply:
I think clarity should win over saving four characters (`vm, `) each time we use it.